sklearn/tests/test_base.py

```diff
@@ -675,3 +675,36 @@ def test_clone_keeps_output_config():
     ss_clone = clone(ss)
     config_clone = _get_output_config("transform", ss_clone)
     assert config == config_clone
+
+
+def test_parent_object_empty_instance_dict():
+    # Since Python 3.11, Python objects have a __getstate__ method by default
+    # that returns None if the instance dict is empty. See #25188.
+    class Empty:
+        pass
+
+    class Estimator(Empty, BaseEstimator):
+        pass
+
+    state = Estimator().__getstate__()
+    expected = {"_sklearn_version": sklearn.__version__}
+    assert state == expected
+
+
+def test_base_estimator_empty_instance_dict():
+    # Since Python 3.11, Python objects have a __getstate__ method by default
+    # that returns None if the instance dict is empty. See #25188.
+
+    # this should not raise
+    state = BaseEstimator().__getstate__()
+    expected = {"_sklearn_version": sklearn.__version__}
+    assert state == expected
```

Comment:
Since we are testing the same behaviour here, I would prefer to parameterise the test and provide the object on which to `__getstate__`.

Also could you put the comment inside a docstring of the test

```python
def test_base_estimator_empty_instance():
    """Check that `__getstate__` returns an empty `dict` with an empty
    instance.

    Python 3.11+ changed behaviour by returning `None` instead of raising
    an `AttributeError`.

    Non-regression test for gh-25188.
    """
    ...
```

I am wondering if we could make the round-trip pickling-unpickling directly in this test instead of having a new test.